### ISSUE TITLE
importlib general type issue

### ISSUE THREAD

**Issue by novasenco** [microsoft/pyright#1946]
**Describe the bug**
erroneous error message for `spec.loader`

**To Reproduce**
```python
from pathlib import PosixPath
from importlib import util

module_file = PosixPath(input("path to python file: "))
module_name = module_file.stem

spec = util.spec_from_file_location(module_name, module_file)
if spec:
    module = util.module_from_spec(spec)
    # Cannot access member "exec_module" for type "_LoaderProtocol" Member "exec_module" is unknown
    spec.loader.exec_module(module) # pyright: reportGeneralTypeIssues=false
    print(module)
```

**Expected behavior**
No error or don't need `# pyright: reportGeneralTypeIssues=false` at end.

**Additional context**
Ps, using pyright via neovim's builtn lsp (with lspconfig).

If this is not a bug, sorry & close.

Thanks

**Comment by erictraut** [comment]
Pyright is doing the right thing here based on the information in the `importlib/abc.pyi` type stub. The `loader` field is annotated with the type `Optional[importlib.abc._LoaderProtocol]`. The class `_LoaderProtocol` has no method called `exec_module`.

If you think that it should have a method called `exec_module`, please consider opening a bug or submitting a PR in the [typeshed](https://github.com/python/typeshed) repo. Pyright consumes these type stubs and relies on them for stdlib types.
